nmostesting/mocks/Registry.py

```diff
@@ -271,3 +417,198 @@ def heartbeat(version, node_id):
+def query(version):
+    registry = REGISTRIES[flask.current_app.config["REGISTRY_INSTANCE"]]
+    if not registry.enabled:
+        abort(503)
+    authorized = registry.check_authorized(request.headers, request.path)
+    if authorized is not True:
+        abort(authorized)
+
+    base_data = ['devices/', 'flows/', 'nodes/', 'receivers/', 'senders/', 'sources/', 'subscriptions/']
+
+    return Response(json.dumps(base_data), mimetype='application/json')
+
+
+def compare_resources(resource1, resource2):
+    try:
+        return NMOSUtils.compare_resource_version(resource1['version'], resource2['version'])
+    except Exception as e:
+        print(e)
+        return 0
+
+
+@REGISTRY_API.route('/x-nmos/query/<version>/<resource>', methods=["GET"], strict_slashes=False)
+def query_resource(version, resource):
+    registry = REGISTRIES[flask.current_app.config["REGISTRY_INSTANCE"]]
+    if not registry.enabled:
+        abort(503)
+    authorized = registry.check_authorized(request.headers, request.path)
+    if authorized is not True:
+        abort(authorized)
+
+    MIN_SINCE = "0:0"
+    MAX_UNTIL = NMOSUtils.get_TAI_time()
+
+    base_data = []
+
+    if request.args.get('paging.since') or request.args.get('paging.until'):
+        registry.pagination_used = True
+
+    since = request.args.get('paging.since') if request.args.get('paging.since') else MIN_SINCE
+    until = request.args.get('paging.until') if request.args.get('paging.until') else MAX_UNTIL
+    new_until = until
+
+    try:
+        valid_resource_types = ['device', 'flow', 'node', 'receiver', 'sender', 'source', 'subscription']
+
+        resource_type = resource.rstrip("s")
+
+        if resource_type not in valid_resource_types:
+            error_message = {"code": 404,
+                             "error": "Invalid resource",
+                             "debug": resource_type}
+            return Response(json.dumps(error_message), status=404, mimetype='application/json')
+
+        registry.query_api_called = True
+
+        # Check to see if resource is being requested as a query
+        if request.args.get('id'):
+            resource_id = request.args.get('id')
+            base_data.append(registry.get_resources()[resource_type][resource_id])
+        else:
+            data = registry.get_resources()[resource_type]
+
+            # only paginate for version v1.1 and up
+            if NMOSUtils.compare_api_version("v1.1", version) > 0:
+                for key, value in data.items():
+                    base_data.append(value)
+            else:
+                data_list = [d for k, d in data.items()]
+                sorted_list = sorted(data_list, key=functools.cmp_to_key(compare_resources))
+
+                result_count = 0
+                for value in sorted_list:
+                    if NMOSUtils.compare_resource_version(value['version'], since) >= 0 \
+                            and NMOSUtils.compare_resource_version(until, value['version']) > 0:
```

Comment:
Hmm, am I mistaken?

```suggestion
                    if NMOSUtils.compare_resource_version(value['version'], since) > 0 \
                            and NMOSUtils.compare_resource_version(until, value['version']) >= 0:
```

Reply:
No you're spot on